src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,687 @@ export default class Fms {
         return waypoints[0];
     }
 
-    // TODO: this should move to a class method
     /**
-     * Returns a flattened array of each `WaypointModel` in the flightPlan
-     *
-     * This is used only in the `CanvasController` when drawing the projected
-     * aircraft path.
+     * Get the next waypoint in the flight plan, if it exists
      *
-     * Using a getter here to stay in line with the previous api.
+     * @for Fms
+     * @property nextWaypoint
+     * @type {WaypointModel}
+     */
+    get nextWaypoint() {
+        return this._routeModel.nextWaypoint;
+    }
+
+    /**
+     * Return an array of all waypoints in all legs of the route
      *
+     * @for Fms
      * @property waypoints
      * @type {array<WaypointModel>}
      */
     get waypoints() {
-        const waypointList = _map(this.legCollection, (legModel) => {
-            return [
-                ...legModel.waypointCollection
-            ];
-        });
-
-        return _flatten(waypointList);
+        return this._routeModel.waypoints;
     }
 
+    // ------------------------------ LIFECYCLE ------------------------------
+
     /**
-     * Initialize the instance and setup initial properties
+     * Initialize class properties
      *
      * @for Fms
      * @method init
      * @param aircraftInitProps {object}
+     * @param navigationLibrary {NavigationLibrary}
+     * @chainable
      */
-    init({ altitude, category, model, route }, initialRunwayAssignment) {
-        this._setCurrentPhaseFromCategory(category);
-        this._setInitialRunwayAssignmentFromCategory(category, initialRunwayAssignment);
-
-        this.flightPlanAltitude = altitude;
+    init({ altitude, category, destination, model, nextFix, origin, routeString }, navigationLibrary) {
```

Comment:
This destructuring should happen on the next line. Doing it here is a little confusing, the method signature should match the comments.

Reply:
done